rfcs/proposed/numa_support/increase-numa-support-availability.org

```diff
@@ -0,0 +1,115 @@
+# -*- fill-column: 80; -*-
+
+#+title: Improve predictability of API for NUMA support
+
+*Note:* This is a sub-RFC of the https://github.com/oneapi-src/oneTBB/pull/1535.
+Specifically, its section about "Increased availability of NUMA support".
+
+* Introduction
+oneTBB has a soft dependency on several variants of ~tbbbind~, which are loaded
+by the library as part of its initialization stage. In turn, each ~tbbbind~ has
+a hard dependency, i.e., relies on load-time linking, on a concrete version of
+the HWLOC library [1, 2]. The soft dependency of oneTBB on ~tbbbind~ allows the
+library to continue its execution even if the system loader is unable to resolve
+the hard dependency on HWLOC for ~tbbbind~. In this case, the HW topology is not
+discovered and the machine is seen as if all CPU cores were uniform, which is
+the default TBB behavior when NUMA constraints are not used. Thus, the following
+code returns meaningless values as these values are just ignored by oneTBB:
+
+#+begin_src C++
+std::vector<oneapi::tbb::numa_node_id> numa_nodes = oneapi::tbb::info::numa_nodes();
+std::vector<oneapi::tbb::core_type_id> core_types = oneapi::tbb::info::core_types();
+#+end_src
+
+An error is also not reported and the client code that uses NUMA support
+facilities may continue running expecting it to work as it was intended. Such
+behavior is not readily noticable by developers that use oneTBB and this
+represents the main problem with the current behavior.
```

Comment:
I keep insisting that the main problem is not the "silence", thus the following re-wording:
```suggestion
This lack of valid HW topology data due to absence of a 3rd party library
is the major problem with the current oneTBB behavior.
There is no diagnostics for the issue, which likely makes it unnoticeable
by developers, and the code that uses oneTBB NUMA support
facilities continues running but does not work as it was intended.
```